### ISSUE TITLE
Investigate why we had to downgrade xhr_headers.rb

### ISSUE THREAD

**Issue by qq99** [Shopify/turbograft#16]
https://github.com/Shopify/turbograft/pull/15

**Comment by qq99** [comment]
@karlhungus can we consider this investigated?  Could you do a little write-up on this?

**Comment by karlhungus** [comment]
>@karlhungus can we consider this investigated? Could you do a little write-up on this?

I think so

Pasted from @Thibaut's comment [here](https://github.com/Shopify/shopify/issues/35973#issuecomment-69348329):

>So the flow of events is:
>- non-Turbolinks request to Rails redirects to `cdn.shopify.com` (or whatever)
>- because of the missing check, `session[:_turbolinks_redirect_to]` is set to `cdn.shopify.com`
>- next (unrelated) Turbolinks request to Rails pops the session and responds with `X-XHR-Redirected-To: cdn.shopify.com`
>- Turbolinks gets the response back and tries to set `cdn.shopify.com` as the current window location, >which isn't allowed (and even if it were, would be wrong)
>
>Only requests coming from Turbolinks should set `session[:_turbolinks_redirect_to]`.

TLDR: we shouldn't have been setting  `session[:_turbolinks_redirect_to]` for every request.

Feel free to reopen if this isn't sufficient :)

**Comment by qq99** [comment]
Thanks :)
